### PR TITLE
Documentation for ablator/modules/metrics

### PR DESCRIPTION
**Pull request by hieuchi911**
Add documentation for `ablator/modules/metrics/main.py` and `ablator/modules/metrics/stores.py`

### CODE REVIEW

File: ablator/modules/metrics/stores.py [hunk 1]
```diff
@@ -13,18 +13,79 @@
...
         batch_limit: int = 30,
         # 100 MB memory limit
         memory_limit: int | None = int(1e8),
     ):
+        """
+        Initialize the storage settings.
+
+        Parameters
+        -----------
+        batch_limit : int, optional
```
Comment (hjzccc) [inline ablator/modules/metrics/stores.py:31]: same as I said in main.py

File: ablator/modules/metrics/stores.py [hunk 2]
```diff
@@ -13,18 +13,79 @@
...
+        ...     batch_limit=10,
+        ...     memory_limit=1000
+        ... )
+        >>> for i in range(100):
+        >>>     array_store.append(int(i))
+        >>> array_store.arr
+        [90, 91, 92, 93, 94, 95, 96, 97, 98, 99]
+        >>> array_store.limit
+        10
+        >>> array_store = ArrayStore(
```
Comment (hjzccc) [inline ablator/modules/metrics/stores.py:85]: maybe you should tell the memory_limit is exceeded at this time

File: ablator/modules/metrics/stores.py [hunk 3]
```diff
@@ -64,6 +161,36 @@ def __init__(
...
     ):
+        """
+        Initialize the storage settings.
+
+        Parameters
+        -----------
+        batch_limit : int, optional
+            Maximum number of batches to keep, so only batch_limit number of latest batches is stored. Default is 30.
+        memory_limit : int or None, optional
+            Maximum memory (in bytes) of batches to keep. Every time this limit is exceeded, batch_limit will be
```
Comment (hjzccc) [inline ablator/modules/metrics/stores.py]: same as I said in main.py

File: ablator/modules/metrics/main.py [hunk 4]
```diff
@@ -31,6 +31,59 @@ def __init__(
         # metrics for which we update with their moving average, i.e. loss
         moving_aux_metrics: Iterable[str] | None = None,
     ):
+        """
+        Initialize the train metrics settings
+
+        Parameters
+        -----------
+        batch_limit : int, optional
```
Comment (hjzccc) [inline ablator/modules/metrics/main.py:39]: this is for every ArrayStore instances in the predictionStore, and I think this description more suitable for a single ArrayStore. Also, the description should be a little different from that in ArrayStore

**Review by hjzccc — COMMENTED** [review]
looks great

File: ablator/modules/metrics/main.py (hunk 4 above)
Comment (hieuchi911) [reply ablator/modules/metrics/main.py:39]: Updated!

File: ablator/modules/metrics/stores.py (hunk 1 above)
Comment (hieuchi911) [reply ablator/modules/metrics/stores.py:31]: Updated!

File: ablator/modules/metrics/stores.py (hunk 2 above)
Comment (hieuchi911) [reply ablator/modules/metrics/stores.py:85]: Updated!

**Review by hjzccc — APPROVED** [review]

**Review by fostiropoulos — APPROVED** [review]
looks great

**Pull request merged**